lib/cretonne/src/isa/mod.rs

```diff
@@ -45,41 +45,83 @@ pub use isa::registers::{RegUnit, RegBank, RegInfo};
 use settings;
 use ir::{InstructionData, DataFlowGraph};
 
+#[cfg(build_riscv)]
 pub mod riscv;
+
+#[cfg(build_intel)]
 pub mod intel;
+
+#[cfg(build_arm32)]
 pub mod arm32;
+
+#[cfg(build_arm64)]
 pub mod arm64;
+
 mod encoding;
 mod enc_tables;
 mod registers;
 
 /// Look for a supported ISA with the given `name`.
 /// Return a builder that can create a corresponding `TargetIsa`.
-pub fn lookup(name: &str) -> Option<Builder> {
+pub fn lookup(name: &str) -> Result<Builder, LookupError> {
     match name {
         "riscv" => riscv_builder(),
         "intel" => intel_builder(),
         "arm32" => arm32_builder(),
         "arm64" => arm64_builder(),
-        _ => None,
+        _ => Err(LookupError::Unknown),
     }
 }
 
 // Make a builder for RISC-V.
-fn riscv_builder() -> Option<Builder> {
-    Some(riscv::isa_builder())
+#[cfg(build_riscv)]
+fn riscv_builder() -> Result<Builder, LookupError> {
+    Ok(riscv::isa_builder())
+}
+
+#[cfg(not(build_riscv))]
+fn riscv_builder() -> Result<Builder, LookupError>{
+    Err(LookupError::Unsupported)
+}
```

Comment:
Yep. This was the reason I split out these little per-target functions. If you can think of a better way of implementing this, I'd be very interested.